### PR TITLE
panel: Read date format from gsettings

### LINKED ISSUE #1039: Enable user selection of date format in time/date settings
The clock on the panel can show the date in the format 'MM/DD/YYYY'. I want to use ISO format but I don't appear to have any way to change the format. Time and Date Settings should allow the user to pick a date format, either with a combo box or a format string like the `date` command uses.

**Operating System**: Solus
**Budgie version**: Budgie 10.3.1

### PR DESCRIPTION
**Pull request by ricardocrudo**
The clock applet read the date format from gsettings instead of use the hardcoded
value. The default format is still '%x'. This closes #1039.

I've placed the format setting at: `/com/solus-project/budgie-panel/date-format`. Let me know if that's not the best path.

### CODE REVIEW

File: src/applets/clock/ClockApplet.vala [hunk 1]
```diff
@@ -301,12 +301,9 @@ public class ClockApplet : Budgie.Applet
...
         }
-        string ftime;
-        if (this.orient == Gtk.Orientation.HORIZONTAL) {
-            ftime = "%x";
-        } else {
-            ftime = "<small>%b %d</small>";
-        }
+
+        settings = new Settings("com.solus-project.budgie-panel");
+        string ftime = (string)settings.get_string("date-format");
```
Comment (ikeydoherty) [inline src/applets/clock/ClockApplet.vala]: Why are you casting a string to a string..?

File: src/applets/clock/ClockApplet.vala [hunk 2]
```diff
@@ -301,12 +301,9 @@ public class ClockApplet : Budgie.Applet
         if (!check_date.get_active()) {
             return;
         }
-        string ftime;
```
Comment (ikeydoherty) [inline src/applets/clock/ClockApplet.vala]: This now breaks vertical panels.

Comment (ricardocrudo) [reply src/applets/clock/ClockApplet.vala]: The customization via gsettings/dconf keeps possible to use the "<small> tag.
If that is not desirable what do you suggest to do? Should I keep it hardcoded when using vertical panel?

**Comment by ikeydoherty** [comment]
I still don't like this, the `small` is enforced for vertical panels and should not be something the user is setting. The string *contents* can come from a setting, sure, but not the style.

**Pull request closed without merging**